### PR TITLE
Use ReflectometrySumInQ instead of local function SumInQ in RRO

### LINKED ISSUE #23011: Use ReflectometrySumInQ in ReflectometryReductionOne
The Q summation done in `ReflectometryReductionOne` is now available as the separate algorithm `ReflectometrySumInQ`. We should be able to remove this code and call `ReflectometrySumInQ` as a child algorithm instead.

However there are a few things to check:
- [ ] #30191 The 2theta pixel size is calculated as the difference between adjacent centres, rather than bounding box size. Our scientists would prefer to continue to use the latter (although it makes little difference with our current instruments, it should technically be more correct). We'd therefore like to add an option to `ReflectometrySumInQ` to use this.
- [ ] #30189 We currently pass `ThetaIn` to provide the horizon angle rather than passing a `BeamCentre` pixel and using its 2theta/2. I think we will need to add a `ThetaIn` property to `ReflectometrySumInQ` (such that either `ThetaIn` or `BeamCentre` must be provided but not both). I don't think we can use `BeamCentre` because we don't have that information available in our workflow (although it could be found through fitting, which we may want to do in future, but we may still want the option to override it).
- [ ] #30192 The `BeamCentre` is currently also used as the reference 2theta that the projection is done onto. This should be arbitrary though so we should be able to use e.g. the centre of the ROI instead if `BeamCentre` is not given.
- [x] #30235 To use a different reference 2theta we'll need to use RefRoi instead of ConvertUnits for the final conversion to Q in RRO
- [ ] #30190 There is an error in the projected wavelength range in RRO which needs to be fixed to give consistent results with RSIQ.

We should also first complete:
- [x] #14000 to use a consistent way of specifying the workspace indices

### PR DESCRIPTION
**Pull request by Pasarus**
**Description of work.**
This is intended to be a move towards making ReflectometryReductionOne a workflow algorithm instead of a more basic algorithm.

**To test:**
- in progress

Fixes #23011

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/ReflectometrySumInQ.cpp [hunk 1]
```diff
@@ -504,19 +531,35 @@ ReflectometrySumInQ::projectedLambdaRange(const MinMax &wavelengthRange,
...
+  } else if (!thetaInProperty->isDefault()) {
+    const int thetaIn = getProperty(Prop::THETA_IN);
+    const double twoTheta = spectrumInfo.twoTheta(static_cast<size_t>(thetaIn));
+    a.horizon = thetaIn;
+    a.twoTheta = twoTheta;
+    a.delta = a.twoTheta - a.horizon;
+    return a;
   } else {
-    a.horizon = 0.;
+    throw std::runtime_error("Neither " + Prop::BEAM_CENTRE + " or " +
```
Comment (soininen) [inline Framework/Algorithms/src/ReflectometrySumInQ.cpp]: This should probably be done in `validateInputs()`.

Comment (Pasarus) [reply Framework/Algorithms/src/ReflectometrySumInQ.cpp]: It is done in validateInputs but, I added an else at the end of a chain of if and else if etc, which I think is in the coding standards....

**Comment by NickDraper** [comment]
@Pasarus This needs some attention.  Antii has left now, so this needs a different reviewer as well.

**Comment by Pasarus** [comment]
I am going to close this PR until further notice.

**Pull request closed without merging**